File: 20220915/모범답안/20220915_04.js/LargestPerimeterTriangle.py
def solution(arr):
    """
    :param arr: int[]
    :return: int
    """

    arr.sort(reverse=True)

    for i in range(len(arr) - 2):
        if isTriangleable(arr[i], arr[i + 1], arr[i + 2]): return arr[i] + arr[i + 1] + arr[i + 2]

    return 0


def isTriangleable(long, mid, short):
    return long < mid + short

File: 20220915/모범답안/20220915_04.js/LargestPerimeterTriangle.js
/**
 * @param arr {number[]}
 * @returns {number}
 */
function solution(arr) {
  arr.sort((a, b) => b - a)

  for (let i = 0; i < arr.length - 2; i++) {
    if (isTriangleable(arr[i], arr[i + 1], arr[i + 2])) return arr[i] + arr[i + 1] + arr[i + 2]
  }

  return 0
}

function isTriangleable(long, mid, short) {
  return long < mid + short
}

solution